### ISSUE TITLE
Dual stacked monitor configuration misposition

### ISSUE THREAD

**Issue by Confucij** [Guake/guake#1162]
I'm using two 1080p monitors placed one over another. And when Guake located on bottom one - which is with a top bar - it's overlapped with a top bar. Details in the attached screenshot.
![screenshot from 2018-01-24 15-57-07](https://user-images.githubusercontent.com/941997/35338802-874753a8-0127-11e8-9d5b-c96f92fb0821.png)

**Comment by Stibbons** [comment]
that will be hard for me to dig it, only have 1 monitor :(

**Comment by Confucij** [comment]
Maybe you could help and point me to code that responsible for window placement and I will dig it by myself?

**Comment by Stibbons** [comment]
guake_app.py function `set_final_window_rect`. It needs a complete rewrite, but if you document well your change and why you needed this change, it will help maintaining it

**Comment by Confucij** [comment]
Thank you. I've managed to hack it for me:
```python
        if window_rect.y != 0:
                window_rect.y = window_rect.y + 25;
```
I'll try to find a time to work on a solution for this.

**Comment by Stibbons** [comment]
hum, this is a generic enough solution, sadly. Where does these 25 pixels comes from? Title bar?

**Comment by Confucij** [comment]
Yep, top panel height. I failed to found any other solution quick. Maybe there is some gtk3 function to find usable monitor space and not just its geometry. 

On StackOverflow, someone suggests create a window, maximize it on a specified monitor, undecorate it, acquire its dimensions and use them as a usable screen space but this is definitely not a 2-minutes-fix.

**Comment by aichingm** [comment]
I'm working on a displacement setting right now which may fix this issue.

**Comment by Confucij** [comment]
@aichingm  Coll! I'm ready to help testing.

**Comment by aichingm** [comment]
@Confucij you can wait for the PR to be merged (#1167) or try it from my master branch at https://github.com/aichingm/guake

**Comment by igabaydulin** [comment]
Does #1167 fix an initial problem?

I have a similar problem. This is how it looks for a single display:
![Screenshot from 2019-07-29 16-30-33](https://user-images.githubusercontent.com/11658656/62052803-f4f57200-b21e-11e9-8a75-6cf68d525c93.png)

This is how it looks for multiple displays:
![Screenshot from 2019-07-29 16-31-13](https://user-images.githubusercontent.com/11658656/62052895-25d5a700-b21f-11e9-968f-7697792d53a9.png)
![Screenshot from 2019-07-29 16-31-19](https://user-images.githubusercontent.com/11658656/62052906-2b32f180-b21f-11e9-94d7-b4681bfb217c.png)

As you can see in both cases the placement is a little bit invalid. When a single display is used the displacement is calculated correctly (regarding launcher and top panel), maybe there is a way to apply displacement automatically for multiple displays too?

#1167 cannot be used when "Appear on mouse display" is enabled:
![Screenshot from 2019-07-29 16-33-55](https://user-images.githubusercontent.com/11658656/62053415-333f6100-b220-11e9-87f1-ad088fcad3ea.png)
![Screenshot from 2019-07-29 16-34-02](https://user-images.githubusercontent.com/11658656/62053424-376b7e80-b220-11e9-9048-af4110b85f99.png)

It applies displacement for both displays, when displacement is needed only for the right one.

Maybe it makes sense to open this issue back?
